rdmo/core/permissions.py

```diff
@@ -85,3 +85,20 @@ class CanToggleElementCurrentSite(DjangoModelPermissions):
     @log_result
     def has_permission(self, request, view):
         return super().has_permission(request, view)
+
+
+class HasRulesPermission(BasePermission):
+
+    @log_result
+    def has_permission(self, request, view) -> bool:
+        if not (request.user and request.user.is_authenticated):
+            return False
+
+        # prefer a view-level override when present
+        perm = getattr(view, 'permission_required', None)
```

Comment:
The `None` is not needed and I would turn it around, check the perm `if perm` and raise the error at the bottom.